packages/twenty-front/src/modules/domain-manager/hooks/useLastAuthenticateWorkspaceDomain.ts

```diff
@@ -0,0 +1,36 @@
+import { useRecoilCallback, useRecoilValue } from 'recoil';
+import { lastAuthenticateWorkspaceDomainState } from '@/domain-manager/states/lastAuthenticateWorkspaceDomainState';
+import { domainConfigurationState } from '@/domain-manager/states/domainConfigurationState';
+import { isDefined } from '~/utils/isDefined';
+
+export const useLastAuthenticateWorkspaceDomain = () => {
+  const domainConfiguration = useRecoilValue(domainConfigurationState);
+
+  const setLastAuthenticateWorkspaceDomain = useRecoilCallback(
+    ({ set }) =>
+      (params: { id: string; subdomain: string } | null) => {
+        set(
+          lastAuthenticateWorkspaceDomainState,
+          isDefined(params)
+            ? {
+                id: params.id,
+                subdomain: params.subdomain,
+                cookieAttributes: {
+                  domain: `.${domainConfiguration.frontDomain}`,
+                },
+              }
+            : null,
+        );
+      },
+    [domainConfiguration],
+  );
+
+  const lastAuthenticateWorkspaceDomain = useRecoilValue(
```

Comment:
don't read lastAuthenticateWorkspaceDomain. we consider states to be part of the api of our modules